### PR TITLE
patch configure to link -lpthread on windows

### PR DESCRIPTION
**Pull request by brendanf**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by brendanf** [comment]
@conda-forge-admin, please rerender

**Comment by brendanf** [comment]
@conda-forge-admin, please rerender

**Comment by brendanf** [comment]
I also opened a PR to patch this upstream https://github.com/r-lib/ps/pull/182

**Comment by brendanf** [comment]
@conda-forge/r ready for review.  Changes welcome.

**Comment by brendanf** [comment]
Fixes #27

**Comment by brendanf** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,6 +26,7 @@ requirements:
     - {{ stdlib("c") }}                # [not win]
     - {{ compiler('m2w64_c') }}        # [win]
     - {{ stdlib("m2w64_c") }}          # [win]
+    - mingw-w64-ucrt-x86_64-winpthreads-git  # [win]
```
Comment (mfansler) [inline recipe/meta.yaml:29]: Likely unnecessary - `libwinpthread` will automatically get pulled in and exported into run requirements. For example, see https://github.com/conda-forge/r-protolite-feedstock/pull/44.

```suggestion
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -38,6 +41,7 @@ requirements:
     - r-base
   run:
     - r-base
+    - mingw-w64-ucrt-x86_64-winpthreads-git  # [win]
```
Comment (mfansler) [inline recipe/meta.yaml:44]:
```suggestion
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for working on this! ❤️ 

As long as the patch gets `-pthread` flag included during link, then the dependencies shouldn't need adjustment.

**Comment by brendanf** [comment]
@mfansler You are right, that worked.

**Review by mfansler — APPROVED** [review]
Thanks again!

**Pull request merged**